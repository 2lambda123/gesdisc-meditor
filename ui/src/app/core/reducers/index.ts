import { createSelector, createFeatureSelector, ActionReducerMap } from '@ngrx/store';
import * as fromContentTypes from './content-types';

export interface ContentTypesState {
  contentTypes: fromContentTypes.State;
}

export interface State {
  contentTypes: ContentTypesState;
}

export const reducers = {
  contentTypes: fromContentTypes.reducer
};

export const selectContentTypesState = createFeatureSelector<ContentTypesState>('contentTypes');


export const getContentTypesEntitiesState = createSelector(
  selectContentTypesState,
  state => state.contentTypes
);

export const {
  selectIds: getContentTypeIds,
  selectEntities: getContentTypeEntities,
  selectAll: getAllContentTypes,
  selectTotal: getTotalContentTypes,
} = fromContentTypes.adapter.getSelectors(getContentTypesEntitiesState);
// export const selectContentTypesIds = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesIds);
// export const selectContentTypesEntities = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesEntities);
// export const selectAllContentTypes = createSelector(selectContentTypesState, fromContentTypes.selectAllContentTypes);
// export const selectContentTypesTotal = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesTotal);
// export const selectCurrentUserId = createSelector(selectUserState, fromContentTypes.getSelectedUserId);

// export const selectCurrentUser = createSelector(
//   selectUserEntities,
//   selectCurrentUserId,
//   (userEntities, userId) => userEntities[userId]
// );
