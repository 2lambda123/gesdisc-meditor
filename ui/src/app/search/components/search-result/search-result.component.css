.author-status {
	margin-bottom: 10px;
}
