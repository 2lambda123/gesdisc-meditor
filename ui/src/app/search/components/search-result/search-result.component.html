<div>
	<a>{{result.title}}</a>
</div>
<div>
	{{result.modifiedOn}} by {{result.modifiedBy}}
</div>
