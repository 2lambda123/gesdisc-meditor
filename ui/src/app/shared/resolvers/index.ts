export { ModelResolver } from './model.resolver'
export { ModelsResolver } from './models.resolver'
export { DocumentResolver } from './document.resolver'
export { DocEditResolver } from './docedit.resolver'
export { AuthGuard } from './auth.guard'
export { RedirectGuard } from './redirect.guard'
