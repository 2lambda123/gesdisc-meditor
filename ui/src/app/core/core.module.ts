import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FlexLayoutModule } from '@angular/flex-layout';
// CONTAINERS
import { MainComponent } from './containers/main';
import { NotFoundPageComponent } from './containers/not-found-page';
import { SplashPageContainer } from './containers/splash-page/splash-page.container';
// COMPONENTS
import { ContentTypeBoxComponent } from './components/content-type-button/content-type-button.component';
import { ContentTypeSplashComponent } from './components/splash-box/splash-box.component';
import { LayoutComponent } from './components/layout';
import { NavItemComponent } from './components/nav-item';
import { SidenavComponent } from './components/sidenav';
import { ToolbarComponent } from './components/toolbar';
import { MaterialModule } from '../material';

import { AngularFontAwesomeModule } from 'angular-font-awesome';

import { AuthModule } from '../auth/auth.module';

export const COMPONENTS = [
  MainComponent,
  NotFoundPageComponent,
  SplashPageContainer,
  ContentTypeBoxComponent,
  ContentTypeSplashComponent,
  LayoutComponent,
  NavItemComponent,
  SidenavComponent,
  ToolbarComponent,
];

@NgModule({
  imports: [CommonModule, RouterModule, MaterialModule, FlexLayoutModule, AngularFontAwesomeModule, AuthModule.forRoot()],
  declarations: COMPONENTS,
  exports: COMPONENTS
})

export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule
    };
  }
}
