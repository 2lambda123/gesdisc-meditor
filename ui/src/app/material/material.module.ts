import { NgModule } from '@angular/core';

import {
  MatInputModule,
  MatCardModule,
  MatButtonModule,
  MatSidenavModule,
  MatListModule,
  MatIconModule,
  MatToolbarModule,
  MatProgressSpinnerModule,
  MatDatepickerModule,
  MatCheckboxModule,
  MatSelectModule
} from '@angular/material';

@NgModule({
  imports: [
    MatInputModule,
    MatCardModule,
    MatButtonModule,
    MatSidenavModule,
    MatListModule,
    MatIconModule,
    MatToolbarModule,
    MatProgressSpinnerModule,
    MatDatepickerModule,
    MatCheckboxModule,
  	MatSelectModule
  ],
  exports: [
    MatInputModule,
    MatCardModule,
    MatButtonModule,
    MatSidenavModule,
    MatListModule,
    MatIconModule,
    MatToolbarModule,
    MatProgressSpinnerModule,
    MatDatepickerModule,
    MatCheckboxModule,
  	MatSelectModule
  ],
})
export class MaterialModule {}
