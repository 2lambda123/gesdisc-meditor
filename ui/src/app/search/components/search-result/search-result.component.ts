import { Component, OnInit, Input } from '@angular/core';
import { DocCatalogEntry } from '../../../service/model/docCatalogEntry';
import { Model } from '../../../service/model/model';

@Component({
	selector: 'med-search-result',
	templateUrl: './search-result.component.html',
	styleUrls: ['./search-result.component.css']
})
export class SearchResultComponent implements OnInit {

	@Input() result: DocCatalogEntry;
	@Input() model: Model;

	constructor() { }

	ngOnInit() {
	}

}
