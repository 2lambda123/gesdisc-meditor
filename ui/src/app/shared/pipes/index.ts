import { NgModule } from '@angular/core';

import { AddCommasPipe } from './add-commas/add-commas';
import { EllipsisPipe } from './ellipsis/ellipsis';
import { EncodeURIComponentPipe } from './encode-uri-component/encode-uri-component'

export const PIPES = [AddCommasPipe, EllipsisPipe, EncodeURIComponentPipe];

@NgModule({
  declarations: PIPES,
  exports: PIPES,
})
export class PipesModule {}
