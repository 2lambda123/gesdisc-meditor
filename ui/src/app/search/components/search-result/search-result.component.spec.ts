import { ComponentFixture, TestBed  } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { SearchResultComponent } from './search-result.component';
import { MatChipsModule } from '@angular/material';
import { PipesModule } from '../../../shared/pipes';

describe('SearchResultComponent', () => {
	let fixture: ComponentFixture<SearchResultComponent>;
	let component: SearchResultComponent;

	beforeEach(() => {
		TestBed.configureTestingModule({
			imports: [
				NoopAnimationsModule,
        MatChipsModule,
        PipesModule
			],
			declarations: [
				SearchResultComponent
			]
		});

		fixture = TestBed.createComponent(SearchResultComponent);
		component = fixture.componentInstance;

	});

	describe('with no data', () => {
		it('should not have any initial input', () => {
			expect(component.result).toBeUndefined();
		});
	});

	describe('with mock data', () => {

		it('should compile with minimum input', () => {
			component.result = {
				title: 'test title',
				'x-meditor': {
					modifiedOn: '2018-05-04T19:09:05.366Z',
					modifiedBy: 'test author',
          state: 'Draft'
				}
			}
			component.model = {
				name: 'test name',
				description: 'test description'
			}

      spyOn(component.loadDocument, 'emit');

      let nativeElement = fixture.nativeElement;
      let doclink = nativeElement.querySelector('a');
      doclink.dispatchEvent(new Event('click'));

			fixture.detectChanges();
			expect(fixture).toMatchSnapshot();
      expect(component.loadDocument.emit).toHaveBeenCalledWith(
        {title: component.result.title, state: component.result['x-meditor'].state}
      );
		});
	});
});
