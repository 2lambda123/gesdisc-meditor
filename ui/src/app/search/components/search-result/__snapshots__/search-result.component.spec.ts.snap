// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultComponent with mock data should compile with minimum input 1`] = `
<med-search-result
  loadDocument={[Function EventEmitter]}
  model={[Function Object]}
  result={[Function Object]}
>
  <div>
    <a
      href="#/document/edit?model=test name&title=test%20title"
    >
       test title 
    </a>
    <mat-chip-list
      aria-disabled="false"
      aria-invalid="false"
      aria-multiselectable="false"
      aria-orientation="horizontal"
      aria-required="false"
      class="state-chip mat-chip-list"
      id="mat-chip-list-1"
      role="listbox"
      tabindex="0"
    >
      <div
        class="mat-chip-list-wrapper"
      >
        <mat-chip
          aria-disabled="false"
          aria-selected="false"
          class="mat-chip mat-primary mat-standard-chip"
          role="option"
          tabindex="-1"
        >
          Draft
        </mat-chip>
      </div>
    </mat-chip-list>
  </div><div
    class="author-status"
  >
     5/4/18, 3:09 PM by test author
  
  </div>
</med-search-result>
`;
